File: cfgs/java_cfgs_foo/FIND_WHETHER_A_GIVEN_NUMBER_IS_A_POWER_OF_4_OR_NOT_1.java
Nodes
1;special;;;
2;statement;int count = 0;14;
3;statement;int x = foo_1 & (foo_2 - 1);15;
4;conditional;(foo_1 > 0) && (foo_2 == 0);16;
5;conditional;foo_1 > 1;17;
6;statement;n >>= 1;18;
7;statement;count += 1;19;
8;statement;return (foo_1 % 2) == 0 ? 1 : 0;21;
9;exit;;;
10;statement;return 0;23;
Edges
0;2;
1;3;
2;4;
3;5;10;
4;8;6;
5;7;
6;5;
7;9;
8;
9;9;

File: cfgs/pythoncfgs/FIND_WHETHER_A_GIVEN_NUMBER_IS_A_POWER_OF_4_OR_NOT_1.py
Nodes
1;statement;foo_1 = 0;8;
2;statement;foo_1 and (not foo_2 & foo_3 - 1);9;
3;conditional;foo_1 > 1;10;
4;conditional;foo_1 % 2 == 0;13;
5;statement;return false;16;
6;statement;return true;14;
7;statement;foo_1 >>= 1;11;
8;statement;foo_1 += 1;12;
9;exit;;
Edges
0;2;
1;3;
2;4;7;
3;5;6;
4;9;
5;9;
6;8;
7;3;
8;
